website/blog/2021-09-06-tungsten.md

```diff
@@ -0,0 +1,358 @@
+we could have just pasted into a braced block and adjust indentation later, but
+now it really matters that indentation is changed properly in order to make to
+code compile correctly.
+
+To help with that [@Giggiux](https://github.com/Giggiux) adjusted the existing
+range formatting to account for indentation in the pasted blocks. Starting from
+this release Metals will adjust the indentation to match that of the enclosing
+block irrelevant of whether it has braces or not. A number of improvements has
+also been done to make sure the cursor ends up on the correct indentation level
+upon writing a newline.
+
+![indentation](https://i.imgur.com/TVewmY7.gif)
+
+If you encounter any issues, please don't hesitate to report them since the
+functionality is new and there might be some edge cases where it doesn't work.
+This was tested internally over the past month, but that doesn't account for all
+the different styles of how people write their code. You can turn off on paste
+formatting in Visual Studio Code in the settings using the
+`editor.formatOnPaste` property.
+
+## New code actions
+
+This release benefits from adding two new code actions. Actions are based on
+particular scenarios, where we might automate some mundane coding and their
+behaviour is usually clearly specified. Please let us know if there are any
+other particular situations, which might be automated, via the the Metals's
+[feature requests repository](http://github.com/metals-feature-requests).
+
+### Refactor pattern match with placeholder
+
+Thanks to [kpodsiad](https://github.com/kpodsiad) it's now possible to
+automatically rewrite lambdas with `_ match` expressions to partial functions:
+
+![partial-func](https://i.imgur.com/Ffm4RXZ.gif)
+
+### Replace () with {} in functions and vice versa
+
+It's quite often that in Scala we might want to change to a simple lambda to a
+multiline one or vice versa. This has previously been a manual chore, but now
+whenever possible you can do it via a code action:
+
+![replace](https://i.imgur.com/DTylGT9.gif)
+
+## Miscellaneous
+
+- Fix completions for definitions from other files in Ammonite scripts.
+- Support for Apple M1 and JDK 17. [@dos65](https://github.com/dos65)
+- [Scala 3] Show extension methods in hover.
+- Support brackets in multiline strings.
+  [@kpodsiad](https://github.com/kpodsiad)
+- Automatically convert $ to $$ when changing to string interpolation.
+- Filter out wrong reference results from unrelated build targets.
+  [@dos65](https://github.com/dos65)
+- Greatly improved parent code lenses performance and turned it on by default in
+  VS Code. [@tanishiking](https://github.com/tanishiking)
+- Properly show folding ranges in Scala 3 indented blocks.
+  [@tpasternak](https://github.com/tpasternak)
+- Fix comments erasing all later variables in .env file.
+- Fix file watcher in large projects on macOS. [@pvid](https://github.com/pvid)
+
```

Comment:
What do you think about mentioning the memory fix that @dos65 did here? I know a lot of people have been mentioning issues with Scala 3 and memory issues, so it might be nice to mention.

Reply:
Added!